analyzers/src/SonarAnalyzer.CSharp/Rules/MemberShouldBeStatic.cs

```diff
@@ -133,16 +133,11 @@ protected override void Initialize(SonarAnalysisContext context)
 
             bool IsExcludedByEnclosingType() =>
                 methodOrPropertySymbol.ContainingType.IsInterface()
-                || (methodOrPropertySymbol.ContainingType.IsGenericType && IsAccessibleOutsideTheType());
-
-            bool IsAccessibleOutsideTheType() =>
-                methodOrPropertySymbol.GetEffectiveAccessibility() switch
-                {
-                    Accessibility.Public => true,
-                    Accessibility.Internal => true,
-                    Accessibility.ProtectedOrInternal => true,
-                    _ => false
-                };
+                || (methodOrPropertySymbol.ContainingType.IsGenericType && IsAccessibleOutsideTheType(methodOrPropertySymbol.GetEffectiveAccessibility()))
```

Comment:
I don't get this: when is `methodOrPropertySymbol.ContainingType.IsGenericType` false and `methodOrPropertySymbol.ContainingType.TypeArguments.Any()` true?

and why not use `methodOrPropertySymbol.GetEffectiveAccessibility()` in both cases? maybe add a comment to explain this, as it's not straight forward